### PR TITLE
Add new recipe for python csaps package

### PR DESCRIPTION
**Pull request by smithsp**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] ~Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).~
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kalling** [comment]
I am confirming that I can be a maintainer.

**Comment by smithsp** [comment]
@conda-forge/help-python, review requested

File: recipes/csaps/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (ocefpaf) [inline recipes/csaps/LICENSE.txt:1]: This file is in the tarball, no need to add it here.

File: recipes/csaps/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{% set name = "csaps" %}
+{% set version = "0.11.0" %}
```
Comment (ocefpaf) [inline recipes/csaps/meta.yaml:2]: Latest version is 1.1.0, right? Any reason why you are adding an old one?

File: recipes/csaps/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+package:
+  name: "{{ name|lower }}"
```
Comment (ocefpaf) [inline recipes/csaps/meta.yaml:5]:
```suggestion
  name: {{ name|lower }}
```

File: recipes/csaps/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  version: "{{ version }}"
```
Comment (ocefpaf) [inline recipes/csaps/meta.yaml:6]:
```suggestion
  version: {{ version }}
```

File: recipes/csaps/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
```
Comment (ocefpaf) [inline recipes/csaps/meta.yaml:9]:
```suggestion
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

File: recipes/csaps/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  sha256: 977644db0aa1a10b9e99c7159ea7f70b064310b8fc4bfd8fc6d293b0bfc3c385
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (ocefpaf) [inline recipes/csaps/meta.yaml:14]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

File: recipes/csaps/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 977644db0aa1a10b9e99c7159ea7f70b064310b8fc4bfd8fc6d293b0bfc3c385
+
+build:
+  number: 0
```
Comment (ocefpaf) [inline recipes/csaps/meta.yaml:13]: Looks like this one can be noarch.
```suggestion
  noarch: python
  number: 0
```

File: recipes/csaps/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - numpy >=1.11.0
+    - pip
+    - python
+    - scipy >=1.0.0
+  run:
+    - {{ pin_compatible('numpy') }}
```
Comment (ocefpaf) [inline recipes/csaps/meta.yaml:23]: I did not check this old version but 1.1.0 doesn't seem to have any compiled extenstions.

```suggestion
    - {{ pin_compatible('numpy') }}
```

File: recipes/csaps/meta.yaml (hunk 2 above)
Comment (smithsp) [reply recipes/csaps/meta.yaml:2]: I am resurrecting this recipe from @kalling , which has worked in the past for us.  I hoped we could merge this version, then let the autotick-bot file a PR for a new version on the new feedstock.

File: recipes/csaps/meta.yaml (hunk 8 above)
Comment (smithsp) [reply recipes/csaps/meta.yaml:23]: Did you mean to remove this line as a suggestion?

File: recipes/csaps/LICENSE.txt (hunk 1 above)
Comment (smithsp) [reply recipes/csaps/LICENSE.txt]: I removed the file.

**Comment by smithsp** [comment]
@ocefpaf Thanks for your prompt review.

File: recipes/csaps/meta.yaml (hunk 8 above)
Comment (ocefpaf) [reply recipes/csaps/meta.yaml:24]: I used grayskull for that version and got:

```
    - python >=3.7,<4.0
    - numpy >=1.11.0,<1.20.0
    - scipy >=1.0.0,<1.6.0
```

Can you adjust that here?

PS: you can do specific versions with `grayskull pypi --strict-conda-forge "csaps=0.11.0"`.

Comment (smithsp) [reply recipes/csaps/meta.yaml]: I regenerated with grayskull and pushed.  Thanks for the pointer about the older version (and `grayskull`).

**Comment by smithsp** [comment]
Thanks @ocefpaf for your help!

**Pull request merged**